patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Inner classes for connection error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
+    def on_reconnected(self, e: ServiceEvent):
+        print("\non_reconnected")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+    
+    def on_reconnecting(self, e: "ServiceEvent"):
+        print("\non_reconnecting")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Inner classes for publish error handling
+class PublisherErrorHandling(PublishFailureListener):
+    def on_failed_publish(self, e: "FailedPublishEvent"):
+        print("on_failed_publish")
+
+# Handle received messages
+class MessageHandlerImpl(MessageHandler):
+    def on_message(self, message: InboundMessage):
+        out_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
+        in_topic = message.get_destination_name()
+        payload_str = message.get_payload_as_string()
+        print(f'Received input message (body string):\n{payload_str}')
+        print(f'Message dump: {message.solace_message.get_message_dump()} \n')
+
+        processed_payload_str = payload_str.upper()
+        print(f'Published output message (body):\n{processed_payload_str}')
+
+        output_msg = msg_builder \
+                        .with_application_message_id(message.get_application_message_id())\
+                        .build(f'{processed_payload_str}')
+
+        direct_publisher.publish(destination=out_topic, message=output_msg)
+
+        print(f'<<<<<<<<<<<<<<<<<<<<<<<<<<<<')
+        print(f'Received input message on {in_topic}')
+        print(f'Received input message (body):\n{payload_str}')
+        print(f'----------------------------')
+        print(f'>>>>>>>>>>>>>>>>>>>>>>>>>>>>')
+        print(f'Published output message on {out_topic}')
+        print(f'Published output message (body):\n{processed_payload_str}')
+        print(f'----------------------------')
```

Comment:
Here is one way to implement the message handler

```
# Handle received messages
class MessageHandlerImpl(MessageHandler):
    def __init__(self, publisher):
        self.publisher = publisher

    def on_message(self, message: InboundMessage):
        payload = message.get_payload_as_string() if message.get_payload_as_string() != None else message.get_payload_as_bytes()
        if isinstance(payload, bytearray):
            print(f"Received a message of type: {type(payload)}. Decoding to string")
            payload = payload.decode()
        
        publish_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
        print(f'Received input message:{payload}')

        processed_payload_str = payload.upper()
        print(f"Processed payload: {processed_payload_str}")
        print(f"Republishing processed message on: {publish_topic.get_name()}")

        # Build the outbound message
        outbound_msg = messaging_service.message_builder() \
                .with_property("application", "processor") \
                .with_property("language", "Python") \
                .build(processed_payload_str)\
        
        # Publish the processed message on the publish_topic
        self.publisher.publish(destination=publish_topic, message=outbound_msg)
```
what do you think?

Reply:
I don't understand what is done in this line:

>         payload = message.get_payload_as_string() if message.get_payload_as_string() != None else message.get_payload_as_bytes()

Please review the changes I made based on the previous comments - it is as good as what you have shown here. If anything missing, we can revise it for sure.

Reply:
This is pretty much how a ternary operator is implemented in python. It's common practice to adopt a ternary operator if there is a one liner if else body. This lines basically reads:
> Set the payload variable to the string representation of the message payload if its not None. otherwise just set it to the byte representation

This is how a ternary operator is set in [Java](https://www.geeksforgeeks.org/java-ternary-operator-with-examples) and [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

Reply:
I propose to keep it as is, and I'll make another PR to change the implementation of all the on_message methods for the message handlers in all the patterns in one PR

Reply:
Got it... it is a weird ternary operator tho :)